src/coreclr/vm/threads.cpp

```diff
@@ -1003,6 +1004,21 @@ HRESULT Thread::DetachThread(BOOL fDLLThreadDetach)
         m_ThreadHandleForClose = hThread;
     }
 
+    if (GCHeapUtilities::IsGCHeapInitialized())
+    {
+        // Guaranteed to NOT be a shutdown case, because we tear down the heap before
```

Comment:
I see you have copied this comment from somwhere else. The comment does not look right. We do not really tear down anything. Most subsystem work till the moment the OS destroys the process.